python/lsst/daf/butler/datastores/fileDatastoreClient.py

```diff
@@ -81,6 +84,8 @@ def get_dataset_as_python_object(
     ]
 
     ref = DatasetRef.from_simple(payload.dataset_ref, universe=universe)
+    if component is not None:
```

Comment:
Maybe note that you have to apply the component override before the storage class override.